### DISCUSSION TITLE
Zoom removing ability to set, refresh and revoke tokens in URL query parameters

### DISCUSSION THREAD

**Discussion by Ralpharama** [Jericho/ZoomNet#250] [General]
I'm sure you are on this but I just thought I'd re-post an email from Zoom I got today about an API update taking effect on the 19th Nov 2022. Just in case ZoomNet sends tokens as query params!
Ralph

---
Zoom API Security Update
 
Dear customer,

As a part of our continuing efforts to improve the security of Zoom OAuth App Types using Authorization flow, we are removing the ability to set the access tokens, refresh tokens and revoke tokens in the URL query parameters. This change will take effect on November 19th, 2022. It is strongly recommended that you update your API calls to set the token values in the Authorization header and not the HTTP query parameters. Failure to do so will result in rejected API calls and a reduced functionality for your application. If you are not the technical point of contact, it is strongly recommended that you forward this message to the appropriate party.

https://marketplace.zoom.us/docs/guides/stay-up-to-date/announcements/#support-for-x-www-form-urlencoded-content-type-headers

**Comment by Jericho** [comment]
I received the same email yesterday, only 5 days before the deadline!

Fortunately, I noticed the announcement last month buried in [this page](https://marketplace.zoom.us/docs/guides/stay-up-to-date/announcements/#support-for-x-www-form-urlencoded-content-type-headers). I implemented the necessary change in #243 and it was released in ZoomNet 0.52.0.

**Reply by Ralpharama** [reply]
Phew! Great thanks, I'd better upgrade to 0.52.0!

**Comment by Jericho** [comment]
Zoom just [announced](https://devforum.zoom.us/t/security-update-no-token-values-in-url-query-parameters/78782/38) they are postponing this breaking change until February 2023. This gives you a few more weeks to upgrade to ZoomNet 0.52.0
